penman/layout.py

```diff
@@ -178,7 +178,8 @@ def _interpret_node(t: Node, variables: Set[Variable], model: Model):
             triples.append(triple)
             epidata.append((triple, epis))
         # nested nodes
-        else:
+        # mypy forgets that (Node ∨ Sym) ^ ¬Sym → Node
+        elif is_tgt_node(target):
```

Comment:
This is pretty awkward, and will technically reduce performance just to get typing to work. Annoyingly, Mypy Typeguard doesn't work the same as `isinstance()` (https://github.com/python/typing/issues/1351). So, replacing `is_tgt_symbol(target)` above with `isinstance(target, str)` then allows Mypy to infer that it must a `Node` here and a `Symbol` above without needing an `elif` here. Not sure which solution is best, since the method is more clear what's going on compared to a naked `isinstance`.

Reply:
Yeah this is a little awkward. In general, I'm not too concerned about reducing performance as long as it's correct.

I haven't looked closely at this code in a while so I'd need to think a bit more about a better solution, but in the meantime I want to point out that the change from `else` to `elif` means that there is no more `else` case. A reader of the code would have to know that `is_tgt_symbol()` and `is_tgt_node()` are defined as opposites to determine that the `elif` would catch all other cases. Otherwise it looks like a latent bug.